tests/unit/state_manager/test_build_state.py

```diff
@@ -117,3 +117,31 @@ def test_property_changes(self, properties):
     def test_project_option_changes(self, project_options):
         state = BuildState(project_options=project_options)
         assert state.diff_project_options_of_interest({}) == set()
+
+    def test_extra_property_changes(self, properties):
+        augmented_properties = {**properties, "extra-property": "foo"}
+        state = BuildState(part_properties=augmented_properties)
+
+        relevant_properties = [
+            "after",
+            "build-attributes",
+            "build-packages",
+            "disable-parallel",
+            "organize",
+            "override-build",
+            "extra-property",
+        ]
+
+        for prop in augmented_properties.keys():
+            other = augmented_properties.copy()
+            other[prop] = "new value"
+
+            diff = state.diff_properties_of_interest(
+                other, also_compare=["extra-property"]
+            )
+            if prop in relevant_properties:
+                # relevant project options changed
+                assert diff == {prop}
```

Comment:
This seems like a case where checking the full difference and providing all failures could lead to a more obvious resolution of any bugs that appear rather than failing on the first problem.

Reply:
In lifecycle context all dirty properties are listed, but the test checks property changes individually to make sure each one can be caught separately. We can add more tests covering different scenarios.